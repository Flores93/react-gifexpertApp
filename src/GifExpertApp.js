import React, { useState } from "react";

import AddCategory from "./components/AddCategory";
import GifGrid from "./components/GifGrid";
const GifExpertApp = () => {
  const categories = ["One Punch"];
  const [cat, setCat] = useState(categories);

  //   const handleAdd = () => {
  //     const newElement = "NewElement";
  //     setCat([...cat, newElement]);
  //   };

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCat={setCat} />
      <hr />

      <ol>
        {cat.map((element) => (
          <GifGrid key={element} category={element} />
        ))}
      </ol>
    </>
  );
};

export default GifExpertApp;
