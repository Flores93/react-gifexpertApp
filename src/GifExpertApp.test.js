import React from "react";
import { shallow } from "enzyme";

import GifExpertApp from "./GifExpertApp";

describe("GifExpertApp tests", () => {
  //Arrange
  let component = shallow(<GifExpertApp />);

  beforeEach(() => {
    component = shallow(<GifExpertApp />);
  });

  test("should show GifExpertApp correctly", () => {
    expect(component).toMatchSnapshot();
  });

  test("should show 'GifExpertApp' inside H2 tag", () => {
    //Act
    const h2Value = component.find("h2").text().trim();

    //Assert
    expect(h2Value).toBe("GifExpertApp");
  });

  //   test("should add a new element in the categories array", () => {
  //     //Act
  //     component.find("button").at(0).simulate("click");

  //     const arr = component.find("li");
  //     console.log(arr.length);

  //     //Assert
  //     expect(arr.length).toBe(4);
  //   });
});
