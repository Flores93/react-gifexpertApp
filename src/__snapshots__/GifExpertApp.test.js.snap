// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp tests should show GifExpertApp correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCat={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
  </ol>
</Fragment>
`;
